interfaces/builtin/registry.go

```diff
@@ -70,8 +70,8 @@ func (iface *registryInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
 	}
 
 	role, ok := plug.Attrs["role"].(string)
-	if !ok || (role != "observer" && role != "manager") {
-		return fmt.Errorf(`registry plug "role" attribute must be "observer" or "manager"`)
+	if ok && role != "manager" {
```

Comment:
Why not allow explicitly setting to "observer"?

Reply:
+1 on that, It seems more natural to make "observer" the default when unset while also allowing setting it explicitly.

Reply:
I don't remember if there was a strong reason to not allow setting "observer" explicitly, can you weigh in @pedronis? I suppose we didn't finish rediscussing roles w/ Gustavo so this might not be final

Reply:
the default is can read and write, but not manage. Observer related to the hooks for observing changes, but for now those are taken care simply by being there or not. So the role itself doesn't make a lot of sense.